src/main/java/creativei/controller/MainController.java

```diff
@@ -17,9 +14,24 @@
 
     @Autowired
     BranchManager branchManager;
+    @Autowired
+    InquiryManager inquiryManager;
 
     @RequestMapping("/")
     public String index() {
         return "Greetings from Spring Boot!";
     }
+
+    @RequestMapping(value="/pincode/beginWith",produces ="application/json",method = RequestMethod.GET)
+    public @ResponseBody
+    ResponseObject getPincodeStartWith( @RequestParam String pincode){
+      ResponseObject responseObject=  inquiryManager.getPincodes(pincode);
+      return responseObject;
+    }
+    @RequestMapping(value = "/pincode/equalsTo",produces = "application/json",method=RequestMethod.GET)
```

Comment:
We are not getting pincode here, but locality data by pincode. Change mapping to /locality/pincode